airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/message/DestinationMessageQueueWriterTest.kt

```diff
@@ -5,40 +5,25 @@
 package io.airbyte.cdk.message
 
 import com.fasterxml.jackson.databind.node.JsonNodeFactory
-import io.airbyte.cdk.command.DestinationCatalog
 import io.airbyte.cdk.command.DestinationStream
 import io.airbyte.cdk.command.MockCatalogFactory.Companion.stream1
 import io.airbyte.cdk.command.MockCatalogFactory.Companion.stream2
 import io.airbyte.cdk.data.NullValue
 import io.airbyte.cdk.state.CheckpointManager
-import io.micronaut.context.annotation.Prototype
+import io.micronaut.context.annotation.Requires
 import io.micronaut.test.extensions.junit5.annotation.MicronautTest
 import jakarta.inject.Inject
-import jakarta.inject.Named
+import javax.inject.Singleton
 import kotlinx.coroutines.test.runTest
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
 
-@MicronautTest(environments = ["MockStreamsManager"])
+@MicronautTest(environments = ["MockStreamsManager"], rebuildContext = true)
 class DestinationMessageQueueWriterTest {
-    @Inject lateinit var queueWriterFactory: TestDestinationMessageQueueWriterFactory
-
-    @Prototype
-    class TestDestinationMessageQueueWriterFactory(
```

Comment:
I wasn't actually sure if this factory was necessary - it seems avoidable, and switching to Singleton+rebuildContext avoids the need for factorying I think?

Reply:
Yes, actually using `@Prototype` also probably made it unecessary to begin with.